AmplifyPlugins/Core/AWSPluginsCore/Model/GraphQLRequest/GraphQLRequest+AnyModelWithSync.swift

```diff
@@ -20,10 +20,11 @@ extension GraphQLRequest {
         documentBuilder.add(decorator: DirectiveNameDecorator(type: .get))
         documentBuilder.add(decorator: ModelIdDecorator(id: id))
         documentBuilder.add(decorator: ConflictResolutionDecorator())
+        documentBuilder.add(decorator: AuthRuleDecorator())
         let document = documentBuilder.build()
 
         return GraphQLRequest<MutationSyncResult?>(document: document.stringValue,
-                                                   variables: document.variables,
+                                                   variables: document.variables.isEmpty ? nil : document.variables,
```

Comment:
hmm why are we having to do this?

Reply:
this is just because the `document.variables` returns [String:Any] that may be empty, and GraphQLRequest contains an optional [String:Any], so to be more accurate the GraphQLRequest should be nil when `document.variables` is empty. There's no real effect on the service by passing in empty variables payload. One thing we can do is to update the variables to be `[String:Any]?` since there is no requirement for `document.variables` to be non-optional, then we can simply pass `document.varibles` to the `GraphQLRequest.variables`

Reply:
i didn't do this becuase it updates all the tests (bad reason lol) but it will be better since it will be aligned with the GraphQLRequest variables

Reply:
made variables optional and updated this to just `variables: document.variables`